### ISSUE TITLE
Synchronization Validation for Presentation

### ISSUE THREAD

**Issue by Eearslya** [KhronosGroup/Vulkan-ValidationLayers#7545]
**Environment:**
 - OS: Windows 10
 - GPU and driver version: NVidia GTX 3080, 551.52
 - SDK or header version if building from repo: 1.3.275.0
 - Options enabled (synchronization, best practices, etc.): Synchronization

**Describe the Issue**

When validating the synchronization between a swapchain image being written to and being presented, the validation layers seem to require a synchronization scope including `VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT`. Without this, the user is given a `PRESENT-AFTER-WRITE` hazard.

However, with the introduction of `VK_KHR_synchronization2`, the top- and bottom-of-pipe flags were deprecated, replaced with `ALL_COMMANDS` or `NONE`.
Per the Vulkan spec:
> `VK_PIPELINE_STAGE_2_BOTTOM_OF_PIPE_BIT` is equivalent to `VK_PIPELINE_STAGE_2_ALL_COMMANDS_BIT` with [VkAccessFlags2](https://registry.khronos.org/vulkan/specs/1.3-extensions/html/vkspec.html#VkAccessFlags2) set to `0` when specified in the first synchronization scope, but equivalent to `VK_PIPELINE_STAGE_2_NONE` in the second scope.

**Expected behavior**

I would expect that including a pipeline barrier or subpass dependency with a `dstStageMask` of `VK_PIPELINE_STAGE_2_NONE` raises no synchronization hazard when used to synchronize an image for presentation, because it is defined to be equivalent to `BOTTOM_OF_PIPE` when used in the second scope.

**Example**

```cpp
// ... acquire swapchain image and perform draw commands ...

VkImageMemoryBarrier2 barrier {VK_STRUCTURE_TYPE_IMAGE_MEMORY_BARRIER_2};
barrier.srcStageMask = VK_PIPELINE_STAGE_2_COLOR_ATTACHMENT_OUTPUT_BIT;
barrier.srcAccessMask = VK_ACCESS_2_COLOR_ATTACHMENT_WRITE_BIT;
barrier.dstStageMask = VK_PIPELINE_STAGE_2_NONE;
barrier.dstAccessMask = VK_ACCESS_2_NONE;

VkDependencyInfo dependency {VK_STRUCTURE_TYPE_DEPENDENCY_INFO};
dependency.imageMemoryBarrierCount = 1;
dependency.pImageMemoryBarriers = &barrier;
vkCmdPipelineBarrier2(commandBuffer, &dependency);

// ... submit command buffer with signal semaphore ...

VkPresentInfoKHR presentInfo {VK_STRUCTURE_TYPE_PRESENT_INFO_KHR};
presentInfo.waitSemaphoreCount = 1;
presentInfo.pWaitSemaphores = &renderFinishedSemaphore;
presentInfo.swapchainCount = 1;
presentInfo.pSwapchains = &swapchain;
presentInfo.pImageIndices = &acquiredImageIndex;
vkQueuePresentKHR(graphicsQueue, &presentInfo);
// Here, a PRESENT-AFTER-WRITE hazard is raised.
// If the barrier is changed to use a dstStageMask of VK_PIPELINE_STAGE_2_BOTTOM_OF_PIPE_BIT, the hazard goes away.
```

**Comment by artem-lunarg** [comment]
@Eearslya that's correct that `BOTTOM_OF_PIPE` in the second scope is equivalent to `NONE`, but because the second scope defines wait operation it means that we wait for nothing in both cases, and `BOTTOM_OF_PIPE` should not be the reason why it worked previously.

The way submission synchronizes with subsequent present is by using a semaphore signaled by submission that is waited by the present. Could you provide your submit setup with a semaphore signal?

**Comment by artem-lunarg** [comment]
The quote from the spec why it's fine to specify no-wait for destination stage before present:
> When transitioning the image to VK_IMAGE_LAYOUT_SHARED_PRESENT_KHR or VK_IMAGE_LAYOUT_PRESENT_SRC_KHR, there is no need to delay subsequent processing, or perform any visibility operations (vkQueuePresentKHR performs automatic visibility operations). To achieve this, the dstAccessMask member of the VkImageMemoryBarrier should be set to 0, and the dstStageMask parameter should be set to VK_PIPELINE_STAGE_BOTTOM_OF_PIPE_BIT.

**Comment by artem-lunarg** [comment]
@Eearslya I realized that the repro case you provided is covered by this fix https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/7480. It was done after SDK 1.3.275 was released. I can confirm it fixes the issue - tested your scenario with SDK 1.3.275 which reports the error, and the same scenario works fine with the latest code. The fix will be included in the next SDK. Another option is to build validation layers from the source code.
